plugins/webservercodes/webserver_codes.go

```diff
@@ -0,0 +1,255 @@
+package webservercodes
+
+import (
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+	"os"
+	"io"
+	"regexp"
+	"errors"
+	
+	"github.com/rogpeppe/rog-go/reverse"
+	"github.com/influxdb/telegraf/plugins"
+)
+
+type Vhost struct {
+    Host string
+    AccessLog string
+    RegexParsestring string
+    Interval string
+}
+
+type Webservercodes struct {
+	Vhosts []*Vhost
+}
+
+type HttpStats struct {
+    codes map[int]int
+}
+
+var sampleConfig = `
+# List of virtualhosts for http codes collecting (each section for one virtualhost, none for disable collecting codes)
+[[webservercodes.vhosts]]
+# 'host' field should match hostname in appropriate status url
+host = "defaulthost"
+
+# telegraf user must have read permissions to this log file (shell command 'sudo adduser telegraf adm' for apache on Ubuntu)
+access_log = "/var/log/apache2/access.log"
+
+# Regular expression for fetching codes from log file strings. You can adjust regex for your custom log format
+# example for apache "common" and "combined" log formats, nginx default log format ("combined"):
+#   '\[(?P<time>[^\]]+)\] ".*?" (?P<code>\d{3})'
+regex_parsestring = '\[(?P<time>[^\]]+)\] ".*?" (?P<code>\d{3})'
+
+# plugin will count http codes in this interval. Interval must be in time.Duration format (see https://golang.org/pkg/time/#ParseDuration)
```

Comment:
okay, now I see that during each collection interval the plugin only counts the status codes in the last `interval` seconds, I initially understood it to be a running counter that would be always-increasing.

In that case keep it the way it is, but think of another name besides `interval`, since throughout telegraf `interval` refers to the collection interval.

maybe call it `logfile_interval` or `code_interval`? Also document this more clearly